src/Product.php

```diff
@@ -1023,6 +1024,33 @@ protected function processTexts(array $data)
         return $this;
     }
 
+    /**
+     * @param array $data
+     * @return $this
+     */
+    protected function processFreeTextFields(array $data)
+    {
+        $freeTexts = array_filter(
+            $data,
+            function($value, $key) {
+                return (bool)preg_match('/^free\d{1,2}$/', $key) && !is_null($value) && $value !== '';
```

Comment:
There's no special API endpoint to get the free fields, they come among other fields, so the regex is used to get the fields that are actually required by this story from the others.
As for why the regex for 'free' and 1-2 digits instead of just checking if the key starts with 'free' - 'free' is a quite common word and I did this as a safeguard in case plentymarkets adds a new field here such as 'freeSomethingId' or similar.

Reply:
I see.. In our Shopware Plugin for example we have a simple for loop.

```php
for ($i = 0; $i <= 20; $i++) {
  $data = $someArray['free' . $i];
}
```

I do not think that plenty will add new free-text fields in the future, the ones they currently have, is already enough. And even if they decide, we could add support for it at any time.